pkg/controller/registry/resolver/util_test.go

```diff
@@ -356,16 +359,107 @@ func NewFakeSourceQuerier(bundlesByCatalog map[CatalogKey][]*api.Bundle) *Namesp
 					return b, nil
 				}
 			}
+
 			return nil, fmt.Errorf("no bundle found")
 		}
+
+		client.FindBundleThatProvidesStub = func(ctx context.Context, groupOrName, version, kind, pkgName string) (*api.Bundle, error) {
```

Comment:
There's a lot to maintain here for a test fake. How about passing an `*api.Bundle` and `error` to `NewFakeSourceQuerier` and returning them here?

Reply:
This `NewFakeSourceQuerier` is used in many places. It is not worth it to change it and have to update the test cases everywhere. The code change is needlessly more. I also prefer we have some actual logic for stubs, not just return what is provided.